pgjdbc/src/test/java/org/postgresql/test/jdbc4/IsValidTest.java

```diff
@@ -67,4 +70,37 @@ public void testTransactionState() throws Exception {
     }
   }
 
+  @Test
+  public void testIsValidRemoteClose() throws Exception {
+
+    assumeMinimumServerVersion("Unable to use pg_terminate_backend before version 8.4", ServerVersion.v8_4);
+
+    Properties props = new Properties();
+    updateProperties(props);
+    Connection con2 = TestUtil.openPrivilegedDB();
+
+    try {
+
+      assertTrue("First Connection should be valid", con.isValid(0));
+
+      String pid;
+      Statement s = con.createStatement();
+
+      try {
+        s.execute("select pg_backend_pid()");
+        ResultSet rs = s.getResultSet();
+        rs.next();
+        pid = rs.getString(1);
+      } finally {
+        TestUtil.closeQuietly(s);
+      }
+
+      TestUtil.execute("select pg_terminate_backend(" + pid + ")", con2);
+
+      assertFalse("The Second connection should now be invalid", con.isValid(0));
```

Comment:
I think that assertion messages could be refined, as `isValid` is still called on _first connection_ as I read it here.

Reply:
Good catch!

Reply:
Actually the code is correct. The first connection is the one that has been terminated.

Reply:
> Actually the code is correct. The first connection is the one that has been terminated.

> that **assertion message** could be refined

IMO it'd be misleading in case of test failure. Message is about _The Second connection_ but the condition is not checked for `con2`, but for *First Connection* (as named in line 84).

> The first connection is the one that has been terminated.

Right, so I do not get why _The Second connection should now be invalid_.

Reply:
Ah, ok, I see what the issue is. The comment... I'll fix. Thanks